### PR TITLE
Update TokenPropertiesRepository to ERC-7508

### PR DESCRIPTION
**Pull request by ThunderDeliverer**
# Description

The token properties repository has been proposed as ERC-7508, and the commit changes reflect the new proposal.

# Checklist

- [x] Verified code additions
- [x] Updated NatSpec comments (if applicable)
- [x] Regenerated docs
- [x] Ran prettier
- [x] Added tests to fully cover the changes

---

### Detailed summary
- Deleted the file `docs/RMRK/extension/tokenProperties/RMRKTokenPropertiesRepository.mdtest/interfaces.ts`.
- Updated the value of `IRMRKTokenPropertiesRepository` to `'0x003f5b32'`.
- Updated the value of `IERC7508` to `'0x07cd44c7'`.
- Updated the `network` value in `hardhat.config.ts` for both `baseGoerli` and `base`.
- Added new methods and their descriptions to the `RMRKTokenAttributesRepository` smart contract.

> The following files were skipped due to too many changes: `docs/RMRK/extension/tokenProperties/RMRKTokenAttributesRepository.md`, `contracts/RMRK/extension/tokenProperties/RMRKTokenPropertiesRepository.sol`, `contracts/RMRK/extension/tokenProperties/IRMRKTokenPropertiesRepository.sol`, `docs/RMRK/extension/tokenProperties/IRMRKTokenPropertiesRepository.md`, `test/extensions/tokenPropertiesRepository.ts`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -46,12 +46,12 @@ Core implementations updates include breaking changes as well; names and symbols
...
-| **Statements** 	|   100.00 %   	|   100.00 %  	|  + 0.00 %  	|
-|  **Branches**  	|    95.53 %   	|   96.78 %   	|  + 1.25 %  	|
-|  **Functions** 	|    99.60 %   	|   99.72 %   	|  + 0.12 %  	|
-|    **Lines**   	|    99.63 %   	|   99.46 %   	|  - 0.17 %  	|
+  | | **Previous** | **Current** | **Change** |
+  |:--------------: |:------------: |:-----------: |------------ |
+  | **Statements** | 100.00 % | 100.00 % | + 0.00 % |
+  | **Branches** | 95.53 % | 96.78 % | + 1.25 % |
+  | **Functions** | 99.60 % | 99.72 % | + 0.12 % |
+  | **Lines** | 99.63 % | 99.46 % | - 0.17 % |
```
Comment (steven2308) [inline CHANGELOG.md:54]: Let's undo this changes.

File: contracts/RMRK/emotable/RMRKEmotesRepository.sol [hunk 2]
```diff
@@ -25,9 +25,9 @@ contract RMRKEmotesRepository is IRMRKEmotesRepository {
         );
 
     // Used to avoid double emoting and control undoing
-    mapping(address emoter => mapping(address collection => mapping(uint256 tokenId => mapping(string emoji => uint256 state))))
+    mapping(address => mapping(address => mapping(uint256 => mapping(string => uint256))))
         private _emotesUsedByEmoter; // Cheaper than using a bool
-    mapping(address collection => mapping(uint256 tokenId => mapping(string emoji => uint256 count)))
+    mapping(address => mapping(uint256 => mapping(string => uint256)))
```
Comment (steven2308) [inline contracts/RMRK/emotable/RMRKEmotesRepository.sol:30]: Let's undo this changes.

File: contracts/RMRK/extension/tokenHolder/RMRKTokenHolder.sol [hunk 3]
```diff
@@ -20,7 +20,7 @@ error InsufficientBalance();
  * @dev The RMRKTokenHolder extension is capable of holding ERC-20, ERC-721, and ERC-1155 tokens.
  */
 abstract contract RMRKTokenHolder is IRMRKTokenHolder {
-    mapping(uint256 tokenId => mapping(address tokenAddress => mapping(TokenType tokenType => mapping(uint256 heldTokenId => uint256 balance))))
+    mapping(uint256 => mapping(address => mapping(TokenType => mapping(uint256 => uint256))))
```
Comment (steven2308) [inline contracts/RMRK/extension/tokenHolder/RMRKTokenHolder.sol:23]: Let's undo this changes.

**Review by steven2308 — APPROVED** [review]
All good beside some things that prettier made ugly :laughing: , thanks for updating ot!

File: contracts/RMRK/emotable/RMRKEmotesRepository.sol (hunk 2 above)
Comment (ThunderDeliverer) [reply contracts/RMRK/emotable/RMRKEmotesRepository.sol:30]: Damn, didn't we fix prettier removing the parameter names? 🤔

**Pull request merged**